template/template.go

```diff
@@ -424,7 +431,11 @@ func (e *executer) resolveSelector(matcher ast.Matcher) ([]any, error) {
 		}
 
 		input, err := e.resolveValue(selectExpr.Operand)
-		if err != nil {
+		if errors.Is(err, mf2.ErrUnresolvedVariable) {
```

Comment:
This does not align with my comment on the ticket - how to handle errors in selectors from the MF2 spec.

Reply:
Should `resolveValue()` return `ErrUnresolvedVariable` instead? Resolving a variable can occur in different places in the MF2. I understand `ErrUnresolvedVariable` should be returned everywhere we try to resolve a variable.